docs/dev/tutorial.md

```diff
@@ -1,125 +1,243 @@
+		//  CheckEmpty checks that there are no extra arguments.
+		return cmd.CheckEmpty(args[1:])
+	}
+}
+
+// Run executes the action of the command. It is part of the Command interface
+// in the juju/cmd package.
+func (c *setQOTDAuthorCommand) Run(ctx *cmd.Context) error {
+	// For now, just tell the user what they wrote.
+	return c.out.Write(ctx, "Quote author set to \""+c.author+"\"")
+}
+```
+
+There are two other methods that are part of the command interface:
+`IsSuperCommand` and `AllowIntersperedFlags`. We do not need to provide these.
+The `setQOTDAuthorComand` struct embeds the `modelcmd.ControllerCommandBase`
+which in several layers down embeds `cmd.Command` from the command package. This
+provides a default implementation.
+
+### Test the command
+
+Juju uses the go check library for testing. To test the command you will add a
+go check unit test suite.
+
+In the `qotd` directory, create a `package_test.go`, and hook up go check to
+work with the `go test` command:
+```go
+package qotd_test
+
+import (
+  stdtesting "testing"
+
+  gc "gopkg.in/check.v1"
+)
+
+func TestPackage(t *stdtesting.T) {
+  gc.TestingT(t)
+}
+```
+Next, in the `qotd` directory, create a `qotd_test.go` file and add a go check
+suite.
+
+```go
+package qotd_test
+
+import (
+  gc "gopkg.in/check.v1"
+
+  "github.com/juju/juju/internal/testing"
+)
+
+type SetQOTDAuthorSuite struct {
+  testing.BaseSuite
+}
+
+var _ = gc.Suite(&SetQOTDAuthorSuite{})
+```
+Embedding the `testing.BaseSuite` provides isolation for the test from the
+system it is running on.
+
+Now, add some tests of the command. We will test the command works as expected
+and returns the right errors.
+
+```go
+func (s *SetQOTDAuthorSuite) TestSetQOTDAuthor(c *gc.C) {
+  context, err := cmdtesting.RunCommand(c, qotd.NewSetQOTDAuthorCommand(), "Nelson Mandela")
+  c.Assert(err, jc.ErrorIsNil)
+  stdout := cmdtesting.Stdout(context)
+  c.Assert(stdout, gc.Equals, `Quote author set to "Nelson Mandela"\n`)
+}
+
+func (s *SetQOTDAuthorSuite) TestSetQOTDAuthorTooManyArgs(c *gc.C) {
+  _, err := cmdtesting.RunCommand(c, qotd.NewSetQOTDAuthorCommand(), "author", "arg-two")
+  c.Assert(err, gc.NotNil)
+  c.Assert(err.Error(), gc.Equals, `unrecognized args: ["arg-two"]`)
+}
+```
+
+**Exercise:** Add a third test to check that the command throws an error when
+no parameters are passed.
+
+> See more: [Go Check](http://labix.org/gocheck)
+
+### Register the command
+The command is implemented, but it needs to be registered to appear when running
+the Juju CLI.
+
+Go to `cmd/juju/command/main.go` and find the `registerCommands` function.
+Import the `qotd` package and register the new command at the bottom.
+```go
+  ...
+
+  // Quote of the day command.
+  r.Register(qotd.NewSetQOTDAuthorCommand())
+}
+```
+
+### Compile and install Juju to try the new command
+From the root of the repo, run:
+```console
+$ make install
+```
+Once it has finished, do `which juju` to check that it has correctly installed.
```

Comment:
Nit: Perhaps can clarify here that "it" refers to the juju binary as there were a few other binaries being installed such as jujuc and jujud, but user should be running the juju binary if they decide to run it from that directory

Reply:
Good point, I've clarified that.